features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/WindowInspector.kt

```diff
@@ -10,54 +10,56 @@ import android.annotation.SuppressLint
 import java.lang.reflect.Field
 
 @SuppressLint("PrivateApi")
 internal object WindowInspector {
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    internal const val FAILED_TO_RETRIEVE_DECOR_VIEWS_ERROR_MESSAGE =
+        "SR WindowInspector failed to retrieve the decor views"
+
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val GLOBAL_WM_CLASS by lazy {
-        try {
-            return@lazy Class.forName("android.view.WindowManagerGlobal")
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy Class.forName("android.view.WindowManagerGlobal")
     }
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val GLOBAL_WM_INSTANCE by lazy {
-        try {
-            return@lazy GLOBAL_WM_CLASS?.getMethod("getInstance")?.invoke(null)
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy GLOBAL_WM_CLASS?.getMethod("getInstance")?.invoke(null)
     }
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val VIEWS_FIELD by lazy {
-        try {
-            return@lazy GLOBAL_WM_CLASS?.getDeclaredField("mViews")
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy GLOBAL_WM_CLASS?.getDeclaredField("mViews")
     }
 
-    fun getGlobalWindowViews(): List<View> {
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            WindowInspector.getGlobalWindowViews()
-        } else {
-            getGlobalWindowViewsLegacy(GLOBAL_WM_INSTANCE, VIEWS_FIELD)
+    @SuppressWarnings("TooGenericExceptionCaught")
+    fun getGlobalWindowViews(internalLogger: InternalLogger): List<View> {
+        return try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                WindowInspector.getGlobalWindowViews()
+            } else {
+                getGlobalWindowViewsLegacy(internalLogger, GLOBAL_WM_INSTANCE, VIEWS_FIELD)
+            }
+        } catch (e: Throwable) {
+            internalLogger.log(
+                InternalLogger.Level.ERROR,
+                InternalLogger.Target.TELEMETRY,
```

Comment:
Same as above, is this actionable? Do we have enough info to diagnose what went wrong?

Reply:
I was thinking that the stacktrace should be enough. We do not send this through telemetry ? Ah damn...I guess not, in that case this is not useful as some others in my code. I thought we are logging also the stacktrace :(. In the same time I need some visibility here as this is quite important, not sure what else I would want to send as there's nothing here. Ideas ?

Reply:
Yup, I think that stacktrace will be enough to get an idea what is going on.